pkg/iac/detection/detect_test.go

```diff
@@ -473,3 +474,140 @@ func BenchmarkIsType_BigFile(b *testing.B) {
 		_ = IsType(fmt.Sprintf("./testdata/%s", "big.file"), bytes.NewReader(data), FileTypeAzureARM)
 	}
 }
+
+func Test_IsFileMatchesSchemas(t *testing.T) {
+
+	schema := `{
+  "$id": "https://example.com/test.schema.json",
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "properties": {
+    "service": { "type": "string" }
+  },
+  "required": ["service"]
+}`
+
+	schema2 := `{
+	"$id": "https://example.com/test.schema.json",
+	"$schema": "https://json-schema.org/draft/2020-12/schema",
+	"type": "object",
+	"properties": {
+	  "provider": { "type": "string" }
+	},
+	"required": ["provider"]
+  }`
+
+	type args struct {
+		schemas     []string
+		fileType    FileType
+		fileName    string
+		fileContent string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		matches bool
+	}{
+		{
+			name: "json file matches",
+			args: args{
+				schemas:  []string{schema},
+				fileType: FileTypeJSON,
+				fileName: "test.json",
+				fileContent: `{
+  "service": "test"
+}`,
+			},
+			matches: true,
+		},
+		{
+			name: "json file dost not matches",
+			args: args{
+				schemas:  []string{schema},
+				fileType: FileTypeJSON,
+				fileName: "test.json",
+				fileContent: `{
+  "somefield": "test",
+}`,
+			},
+			matches: false,
+		},
+		{
+			name: "json file matches, but file type is yaml",
+			args: args{
+				schemas:  []string{schema},
+				fileType: FileTypeYAML,
+				fileName: "test.json",
+				fileContent: `{
+  "service": "test"
+}`,
+			},
+			matches: false,
+		},
+		{
+			name: "broken json file",
+			args: args{
+				schemas:  []string{schema},
+				fileType: FileTypeJSON,
+				fileName: "test.json",
+				fileContent: `{
+  "service": "test",,
+}`,
+			},
+			matches: false,
+		},
+		{
+			name: "yaml file matches",
+			args: args{
+				schemas:     []string{schema},
+				fileType:    FileTypeYAML,
+				fileName:    "test.yml",
+				fileContent: `service: test`,
+			},
+			matches: true,
+		},
+		{
+			name: "yaml file dost not matches",
+			args: args{
+				schemas:     []string{schema},
+				fileType:    FileTypeYAML,
+				fileName:    "test.yaml",
+				fileContent: `somefield: test`,
+			},
+			matches: false,
+		},
+		{
+			name: "broken yaml file",
+			args: args{
+				schemas:  []string{schema},
+				fileType: FileTypeYAML,
+				fileName: "test.yaml",
+				fileContent: `text foobar
+number: 2`,
+			},
+			matches: false,
+		},
+		{
+			name: "multiple schemas",
+			args: args{
+				schemas:     []string{schema, schema2},
+				fileType:    FileTypeYAML,
+				fileName:    "test.yaml",
+				fileContent: `provider: test`,
+			},
+			matches: true,
+		},
```

Comment:
What is the behavior if one schema matched and the other didn't?

Reply:
A file is suitable for scanning if it matches at least one schema.